### ISSUE TITLE
apiset still used in S.R.Metadata

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/coreclr#10974]
@yizhang82 @stephentoub @tmat, @nguerrera @gkhanna79 is there any reason to still use apisets in this assembly? We removed them in all other places. But perhaps we need to keep it in this library because it ships OOB and does not necessarily have reverse forwarders?

```c#
// The library guards against unavailable entrypoints by using EntryPointNotFoundException.
        private static unsafe class NativeMethods
        {
            // API sets available on modern platforms:
            [DllImport(@"api-ms-win-core-file-l1-1-0.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
            [return: MarshalAs(UnmanagedType.Bool)]
            internal static extern bool ReadFileModern(
                 SafeHandle fileHandle,
                 byte* buffer,
                 int byteCount,
                 out int bytesRead,
                 IntPtr overlapped
            );

            // older Windows systems:
            [DllImport(@"kernel32.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
            [return: MarshalAs(UnmanagedType.Bool)]
            internal static extern bool ReadFileCompat(
                 SafeHandle fileHandle,
                 byte* buffer,
                 int byteCount,
                 out int bytesRead,
                 IntPtr overlapped
            );
        }
#pragma warning restore BCL0015
```

**Comment by danmosemsft** [comment]
Either way it's wrapped in `#pragma warning disable BCL0015 // Diasable Pinvoke analyzer errors.`. 

For some reason the Kernel32 one is not in our OneCore list (hence suppression) but is in the UWP list (WACK always allows apiset and classic versions). Need to make certain, but I believe this means we're OK for UWP though.

**Comment by yizhang82** [comment]
These are identical. We would only need the kernel32 version.

**Comment by danmosemsft** [comment]
OK we should clean that up at some point.

**Comment by gkhanna79** [comment]
Any reason why this cannot be fixed in 2.0, as opposed to 2.1.0?

**Comment by tmat** [comment]
Jan is making some changes: https://github.com/dotnet/corefx/pull/18356.

/cc @jkotas

**Comment by jkotas** [comment]
I have included the fix for this one in my PR.

**Comment by jkotas** [comment]
This issue was moved to dotnet/corefx#18387

**Comment by danmosemsft** [comment]
Oh doh. Wrong window thx.

**Comment by danmosemsft** [comment]
> Any reason why this cannot be fixed in 2.0, as opposed to 2.1.0?
Milestone=2.0 is for issues that must be fixed to ship 2.0 per our current bar. It is not for cleanup work unless that meets that bar. Of course folks can still commit stuff that doens't meet the bar, until we fork.
